Core/TokenServices/TokenGenerationServices/TokenGeneration.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Data.AppDbContext;
+using Data.Models;
+
+
+namespace Core.TokenServices.TokenGenerationService
+{
+    public class TokenGenerationService : ITokenGenerationService
+    {
+        private readonly TeejayDbContext _context;
+        private static readonly Random _random = new Random();
+
+        public TokenGenerationService(TeejayDbContext context)
+        {
+            _context = context;
+        }
+
+        public List<Token> GenerateTokens(int numberOfTokens)
```

Comment:
The Return type should be of type IEnumerable and not List. IEnumerable is to be used because we intend not to modify the data returned, hence it is of "read only" type. That is why we need to use IEnumerable.